docs/python-support.md

```diff
@@ -27,13 +24,16 @@ The Python support generated by Djinni makes use of a few non-default Python ext
 
 ## Test Suite
 
-Because of the known limitations described below, the Python test suite does not use all of the same inputs as the other languages.  As a result, you'll see a separate `pycpp` directory in generated-src.  This contains output of the standard C++ generator, but for the specific input files used by the Python test suite.
+Because of the known limitations described below, the Python test suite does not use all of the same
+inputs as the other languages.  As a result, you'll see a separate `pycpp` directory in
+generated-src.  This contains output of the standard C++ generator, but for the specific input files
+used by the Python test suite.
 
 ## Known limitations
 
-* Proxy caching is not yet implemented.  In addition to some inefficiency, this means that the same object passed across the language boundary twice is not guaranteed to appear as a single object on the other side.
-* External types defined in YAML are not yet supported.
-* The support library code makes direct reference to <optional> types rather than respecting the command-line choice of a different optional library.
-* Use of non-nullable pointers is not yet supported.
-* There are no arguments for configurable header prefixes (or use of existing arguments).  All includes are unqualified.
+* Proxy caching is not yet implemented.  In addition to some inefficiency, this means that the same
```

Comment:
Are you sure that proxy-caching is not implemented? I've watched the talk from CppCon 2015, and one aspect they were skimming was that they had worked on a unified proxy cache mechanism when introducing Python support. 

[quote](https://youtu.be/K-k-axW2utc?t=2456)
> We used to have 4 seperate [Cache] implementations between each of Java, C++, Obj-C, each direction. When I started to implement this for Python, we realized: [...] Let's not go from 4 to 6, let's go from 4 to 1! So now this actually shares code...

I haven't digged into the code to proof that, I just wanted to share my doubts. 

@artwyman can you help with that?

Reply:
That does seem inconsistent, but it's been too long for me to remember the details.  Poking around the code back in the original Dropbox repo, I think we did do the refactor to make a unified proxy cache template (see proxy_cache_interface.hpp).  However I can't find any sign of it being used in the generated code for Python.  Searching for `proxy_cache` in the repo I only find those headers referenced in the jni and objc support libs.

Also here's an example of generated code where use of the proxy cache would be expected to start: https://github.com/dropbox/djinni/blob/python/test-suite/generated-src/cwrapper/cw__foo_listener_bf.cpp#L48

There are a lot of layers of twisty code there, but it looks to me like they're all dealing with wrapping/unwrapping smart pointers and ref-counting, but no reference to the global proxy cache.

I suspect what happened was that Python motivated us to do the unification, but the refactor was done by a different developer.  The intern who was working on Python didn't get the chance to integrate with it before the end of the summer.

Reply:
You could prove it for sure by writing a test, which would also act as as an acceptance test when the caching is implemented.  The point of the caching is that when you pass the same object through the boundary twice (while the first reference is still alive) you get the same result.  I.e. you can directly compare two pointers for equality (== in C++, `is` in Python) and see that they're the same object.  I don't have the right dev environment set up to write and run that test right now, but it should be pretty simple to set up if you do.